jobs: factor out jobspb package

tbg opened the pull request:
Lots of stuff depends on `jobs` and `jobs` depends on lots of stuff.
This makes it hard to import that package, as most recently seen in
\#26551. Introduce a `jobspb` package instead.

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/26664)
<!-- Reviewable:end -->

benesch commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo the stuff you missed in the UI that lint is complaining about

---

Reviewed 30 of 30 files at r1.
Review status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26664#-:-LEqgI2tBvE_PXx3UTL9:buou69w)*
<!-- Sent from Reviewable.io -->

tbg committed:
jobs: factor out jobspb package

Lots of stuff depends on `jobs` and `jobs` depends on lots of stuff.
This makes it hard to import that package, as most recently seen in
\#26551. Introduce a `jobspb` package instead.

Release note: None.

tschottdorf commented:
bors r=mjibson,benesch

tschottdorf commented:
bors r=bdarnell,benesch

Flake was TestChangefeedPauseUnpause which has since been skipped.
